programs/stake/src/stake_state.rs

```diff
@@ -287,20 +295,31 @@ fn calculate_stake_rewards(
 
     // Drive credits_observed forward unconditionally when rewards are disabled
     // or when this is the stake's activation epoch
-    if point_value.rewards == 0
-        || (fix_activating_credits_observed && stake.delegation.activation_epoch == rewarded_epoch)
```

Comment:
I'm intentionally removing this obsolete feature guard ( `fix_activating_credits_observed` ) because i believe this has been activated across board already (mainnet-beta, testnet, devnet). CC: @joncinque 

On the other hand, i'm trying to reuse its plumbing for another my feature (ref: #22546 ). so, I'm keeping the plumbing intentionally as well.

Reply:
Yep, it's been active for awhile, so feel free to remove it